aclmapping/bank/mapping_test.go

```diff
@@ -0,0 +1,88 @@
+package aclbankmapping
+
+import (
+	"testing"
+
+	pp "github.com/k0kubun/pp/v3"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+
+func cacheTxContext(ctx sdk.Context) (sdk.Context, sdk.CacheMultiStore) {
+	ms := ctx.MultiStore()
+	// TODO: https://github.com/cosmos/cosmos-sdk/issues/2824
+	msCache := ms.CacheMultiStore()
+	return ctx.WithMultiStore(msCache), msCache
+}
+
+
+func TestSendToModuleAccount(t *testing.T) {
+	priv1 := secp256k1.GenPrivKey()
+	addr1 := sdk.AccAddress(priv1.PubKey().Address())
+	priv2 := secp256k1.GenPrivKey()
+	addr2 := sdk.AccAddress(priv2.PubKey().Address())
+	coins     := sdk.Coins{sdk.NewInt64Coin("foocoin", 10)}
+
+	tests := []struct {
+		name          string
+		expectedError error
+		msg           *types.MsgSend
+	}{
+		{
+			name:          "default send",
+			msg:           types.NewMsgSend(addr1, addr2, coins),
+			expectedError: nil,
+		},
+	}
+
+	acc1 := &authtypes.BaseAccount{
+		Address: addr1.String(),
+	}
+	acc2 := &authtypes.BaseAccount{
+		Address: addr2.String(),
+	}
+	accs := authtypes.GenesisAccounts{acc1, acc2}
+	balances := []types.Balance{
+		{
+			Address: addr1.String(),
+			Coins:   coins,
+		},
+		{
+			Address: addr2.String(),
+			Coins:   coins,
+		},
+	}
+
+	app := simapp.SetupWithGenesisAccounts(accs, balances...)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+
+	pp.Default.SetColoringEnabled(false)
+	handler := bank.NewHandler(app.BankKeeper)
+
+	for _, tc := range tests {
+
+		t.Run(tc.name, func(t *testing.T) {
+			handlerCtx, cms := cacheTxContext(ctx)
+			_, err := handler(handlerCtx, tc.msg)
+			depdenencies , _ := MsgSendDependencyGenerator(app.AccessControlKeeper, handlerCtx, tc.msg)
+
+			pp.Println(cms.GetEvents())
```

Comment:
Events changes in this PR:
https://github.com/sei-protocol/sei-cosmos/pull/70/files